### PR TITLE
semver-tool: init at 2.1.0

### PR DESCRIPTION
**Pull request by alyssais**
###### Motivation for this change

Requested by @coretemp in https://github.com/NixOS/nixpkgs/issues/50945.

I chose "semver-tool" as the path for this package because "semver" is extremely generic, and probably more rightly belongs to [node-semver][1] (at least judging by GitHub stars), and because "semver-tool" is the name of this project's GitHub repository, and because it was the name used in the [package request][2].

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

[1]: https://github.com/npm/node-semver
[2]: https://github.com/NixOS/nixpkgs/issues/50945

### CODE REVIEW

**Comment by worldofpeace** [comment]
```
test/documentation-test
/nix/store/n2hjbpkf4c0m48945ivxs3lmsczzw2rg-bash-4.4-p23/bin/bash: test/documentation-test: /usr/bin/env: bad interpreter: No such file or directory
make: *** [Makefile:4: test] Error 126
```

Probably need to have `patchShebangs` ran earlier.

File: pkgs/development/tools/misc/semver-tool/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  version = "2.1.0";
+
+  src = fetchFromGitHub {
+    owner = "fsaintjacques";
+    repo = "semver-tool";
+    rev = version;
+    sha256 = "0lpwsa86qb5w6vbnsn892nb3qpxxx9ifxch8pw3ahzx2dqhdgnrr";
+  };
+
+  installPhase = ''
```
Comment (jtojnar) [inline pkgs/development/tools/misc/semver-tool/default.nix]: Please add `runHook preInstall` and `runHook postInstall`. And also report the issue with path hardcoded in the install path.

**Comment by alyssais** [comment]
Hardcoded prefix reported upstream: https://github.com/fsaintjacques/semver-tool/pull/24

File: pkgs/development/tools/misc/semver-tool/default.nix (hunk 1 above)
Comment (alyssais) [reply pkgs/development/tools/misc/semver-tool/default.nix]: Oh, glad to see that's something we're caring about now :)

File: pkgs/development/tools/misc/semver-tool/default.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+  src = fetchFromGitHub {
+    owner = "fsaintjacques";
+    repo = "semver-tool";
+    rev = version;
+    sha256 = "0lpwsa86qb5w6vbnsn892nb3qpxxx9ifxch8pw3ahzx2dqhdgnrr";
+  };
+
+  postPatch = ''
+    patchShebangs .
```
Comment (jtojnar) [inline pkgs/development/tools/misc/semver-tool/default.nix]: You should only patch the tests – this will patch `src/semver` with build-time bash as well.

Comment (alyssais) [reply pkgs/development/tools/misc/semver-tool/default.nix]: Ah, didn't occur to me that build-time bash could be different from runtime. But of course they can (eg if crosscompiling, right?).

**Pull request closed without merging**